packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm

```diff
@@ -15,6 +15,7 @@
 #import "RCTLinkingPlugins.h"
 
 static NSString *const kOpenURLNotification = @"RCTOpenURLNotification";
+static NSURL *initialURL = nil;
```

Comment:
To investigate: What are the downsides of using static property here? This is the cleanest solution that I found (which works) and it doesn't require to change upstream interfaces